AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs

```diff
@@ -24,6 +24,28 @@ public class AzureCSharpCodeNamer : CSharpCodeNamer
             // Do nothing
         }
 
+        public AzureCSharpCodeNamer(Settings settings)
+        {
+            if (settings != null && settings.CustomSettings != null)
```

Comment:
This is to fix the bug where the base class' property: UseDateTimeOffset is not being set when the AzureCSharpCodeNamer is instantiated. We don't care if settings are passed in, so this constructor now behaves exactly like the default constructor if settings is null, however if settings is not null we will attempt to properly set this value.